Source/dthread.cpp

```diff
@@ -78,95 +76,96 @@ void __fastcall dthread_send_delta(int pnum, char cmd, void *pbSrc, int dwLen)
+		TermMsg("dthread:1\n%s", error_buf);
+	}
+
+	byte_52A508 = 1;
+
+	sghThread = (HANDLE)_beginthreadex(NULL, 0, dthread_handler, NULL, 0, &glpDThreadId);
+	if (sghThread == (HANDLE)-1) {
+		error_buf = TraceLastError();
+		TermMsg("dthread2:\n%s", error_buf);
 	}
 }
-// 52A508: using guessed type char byte_52A508;
-// 679660: using guessed type char gbMaxPlayers;
 
-unsigned int __stdcall dthread_handler(void *a1)
+unsigned int __stdcall dthread_handler(void *unused)
 {
-	char *v1;        // eax
-	TMegaPkt *v2;    // esi
-	int v3;          // ecx
-	unsigned int v4; // edi
+	char *error_buf;
+	TMegaPkt *pkt;
+	DWORD dwMilliseconds;
 
 	while (byte_52A508) {
 		if (!sgpInfoHead && WaitForSingleObject(sghWorkToDoEvent, 0xFFFFFFFF) == -1) {
-			v1 = TraceLastError();
-			TermMsg("dthread4:\n%s", v1);
+			error_buf = TraceLastError();
+			TermMsg("dthread4:\n%s", error_buf);
 		}
+
 		EnterCriticalSection(&sgMemCrit);
-		v2 = sgpInfoHead;
+		pkt = sgpInfoHead;
 		if (sgpInfoHead)
 			sgpInfoHead = sgpInfoHead->pNext;
 		else
 			ResetEvent(sghWorkToDoEvent);
 		LeaveCriticalSection(&sgMemCrit);
-		if (v2) {
-			v3 = v2->dwSpaceLeft;
-			if (v3 != 4)
-				multi_send_zero_packet(v3, v2->data[0], &v2->data[8], *(_DWORD *)&v2->data[4]);
-			v4 = 1000 * *(_DWORD *)&v2->data[4] / (unsigned int)gdwDeltaBytesSec;
-			if (v4 >= 1)
-				v4 = 1;
-			mem_free_dbg(v2);
-			if (v4)
-				Sleep(v4);
+
+		if (pkt) {
+			if (pkt->dwSpaceLeft != 4)
+				multi_send_zero_packet(pkt->dwSpaceLeft, pkt->data[0], &pkt->data[8], *(_DWORD *)&pkt->data[4]);
+
+			dwMilliseconds = 1000 * *(_DWORD *)&pkt->data[4] / gdwDeltaBytesSec;
+			if (dwMilliseconds >= 1)
+				dwMilliseconds = 1;
+
+			mem_free_dbg(pkt);
+
+			if (dwMilliseconds)
+				Sleep(dwMilliseconds);
 		}
 	}
+
 	return 0;
 }
-// 52A508: using guessed type char byte_52A508;
 // 679730: using guessed type int gdwDeltaBytesSec;
 
 void __cdecl dthread_cleanup()
 {
-	char *v0;     // eax
-	TMegaPkt *v1; // eax
-	TMegaPkt *v2; // esi
-
-	if (sghWorkToDoEvent) {
-		byte_52A508 = 0;
-		SetEvent(sghWorkToDoEvent);
-		if (sghThread != (HANDLE)-1 && glpDThreadId != GetCurrentThreadId()) {
-			if (WaitForSingleObject(sghThread, 0xFFFFFFFF) == -1) {
-				v0 = TraceLastError();
-				TermMsg("dthread3:\n(%s)", v0);
-			}
-			CloseHandle(sghThread);
-			sghThread = (HANDLE)-1;
-		}
-		CloseHandle(sghWorkToDoEvent);
-		v1 = sgpInfoHead;
-		sghWorkToDoEvent = 0;
-		if (sgpInfoHead) {
-			do {
-				v2 = v1->pNext;
-				sgpInfoHead = 0;
-				mem_free_dbg(v1);
-				v1 = v2;
-				sgpInfoHead = v2;
-			} while (v2);
+	char *error_buf;
+	TMegaPkt *tmp1, *tmp2;
+
+	if (sghWorkToDoEvent == NULL) {
+		return;
+	}
+
+	byte_52A508 = 0;
+	SetEvent(sghWorkToDoEvent);
+	if (sghThread != (HANDLE)-1 && glpDThreadId != GetCurrentThreadId()) {
+		if (WaitForSingleObject(sghThread, 0xFFFFFFFF) == -1) {
+			error_buf = TraceLastError();
+			TermMsg("dthread3:\n(%s)", error_buf);
 		}
+		CloseHandle(sghThread);
+		sghThread = (HANDLE)-1;

```

Comment:
Do we have the `INVALID_HANDLE` define or was that in later versions of `windows.h`?

Reply:
`INVALID_HANDLE_VALUE` it turns out